src/cargo/core/compiler/fingerprint.rs

```diff
@@ -1108,13 +1108,8 @@ fn calculate_normal<'a, 'cfg>(
     })
 }
 
-// We want to use the mtime for files if we're a path source, but if we're a
-// git/registry source, then the mtime of files may fluctuate, but they won't
-// change so long as the source itself remains constant (which is the
-// responsibility of the source)
 fn use_dep_info(unit: &Unit<'_>) -> bool {
-    let path = unit.pkg.summary().source_id().is_path();
-    !unit.mode.is_doc() && path
+    !unit.mode.is_doc()
```

Comment:
Could this perhaps be a separate change? This is something I'd want to take more time to dig into the source and figure out what's going on and what the implications of this change are

Reply:
Yep, I'll split this out. I agree that the implications are perhaps not entirely clear -- let me know if it'd be helpful for me to do some digging and provide a report of some sort.

Reply:
I'm basically curious about the performance impact of this and how many locations call `use_dep_info`. I'm worried, for example, that we're going to try to open a git repository all over the place and cause issues. I don't mind looking into this though to see what's going on.